src/core/email.py

```diff
@@ -315,3 +310,32 @@ def get_email_content(request, setting_name, context):
     html_content = htmly.render(con)
 
     return html_content
+
+
+def get_email_greeting(recipients, adjective='Dear'):
+    """Composes an email greeting to a list of Users.
+
+    Args:
+        recipients (list): ordered enumerable containing Users.
+        adjective (str): the leading adjective to the greeting.
+
+    Returns (str): an email greeting to one or more users.
+
+    """
+    recipient_forenames = [recipient.first_name for recipient in recipients]
+
+    recipients_greeting_parts = []
+    for i, forename in enumerate(recipient_forenames):
+        if i == 1:
+            recipients_greeting_parts.append(' and ')
+        elif i > 1:
+            recipients_greeting_parts.append(', ')
+        recipients_greeting_parts.append(forename)
+
+    recipients_greeting_parts.append(
+        '{adjective} '.format(
+            adjective=adjective
+        )
+    )
+    # The greeting is composed in reverse for simplicity
```

Comment:
Actually, ignore this. It won't work properly for a single user

Reply:
Thanks for this - very elegant!

To avoid it joining `''` and `'Jane'` with `', '` in that case of `recipient_forenames = ['Jane']` I had to make it a bit 
 awkward by adding a list comprehension with a conditional:
```
    return '{adjective} {recipients}'.format(
         adjective=adjective,
         recipients=', '.join(
             greeting_substring for greeting_substring in
             [', '.join(recipient_forenames[:-2])] +
             [' and '.join(recipient_forenames[-2:])]
             if greeting_substring
         )
    )
```
Still far better than the original.

If `join` was okay with concatenating a mixture of strings and lists I could just use `or`

Reply:
Oh, I've just seen your other comment. I'd be happy to use `filter`, I think, it's still more readable than my first attempt.

Reply:
Nicely done

Reply:
Yes, sorry I edited the comment. Either way works fine, so whatever you are happy with